packages/dom/src/dom/is-number-input.js

```diff
@@ -2,12 +2,16 @@
  * Check whether the given element is an input field of type number
  * and has a valueAsNumber
  *
- * @param {HTMLElement} element The HTML element.
+ * @param {Element} element The HTML element.
  *
  * @return {boolean} True if the element is input and holds a number.
```

Comment:
We've got the same type guard as `isHTMLInputElement` here, let's take advantage:
```suggestion
 * @return {element is HTMLInputElement} True if the element is input and holds a number.
```